src/ATokenCheckpointed.sol

```diff
@@ -0,0 +1,307 @@
+// SPDX-License-Identifier: UNLICENSED
```

Comment:
This is very similar to the existing [ATokenNaive](https://github.com/ScopeLift/flexible-voting/blob/1cd6967167928a48deb796c2c8f1a2d820a70d31/src/ATokenNaive.sol) implementation. Here's a diff of the two: https://www.diffchecker.com/WZ9bQlTv

In case that link expires, it's just [this code](https://raw.githubusercontent.com/ScopeLift/flexible-voting/7efdd7b8deabd91a4950fc02ab6ef659bac32156/src/ATokenCheckpointed.sol) diff'ed against [this code](https://raw.githubusercontent.com/ScopeLift/flexible-voting/1cd6967167928a48deb796c2c8f1a2d820a70d31/src/ATokenNaive.sol)